sharedbuffers/mapped_struct.py

```diff
@@ -864,6 +886,72 @@ def __getitem__(self, index):
 
         return res
 
+    @staticmethod
+    def _adjust_min(val, nmax, step):
+        val = min(val, nmax)
+        if step < 0 and val == nmax:
+            val -= 1
+        return val
+
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            start, end, step = index.start, index.stop, index.step
+            xlen = len(self)
```

Comment:
To make this more readable, why don't  you move this body to a `_getslice` method?